src/parse.c

```diff
@@ -1112,12 +1112,66 @@ handle_addresses(yaml_document_t* doc, yaml_node_t* node, const void* _, GError*
     return handle_generic_addresses(doc, node, TRUE, &(cur_netdef->ip4_addresses), &(cur_netdef->ip6_addresses), error);
 }
 
+struct _gw_parse_result {
+    const char *gw4;
+    const char *gw6;
+};
+
+static gboolean
+handle_single_gateway(yaml_node_t* node, GError** error, struct _gw_parse_result *out)
+{
+    const char *addr;
+
+    assert_type(node, YAML_SCALAR_NODE);
+
+    addr = scalar(node);
+    if (is_ip4_address(addr)) {
+        if (out->gw4)
+            return yaml_error(node, error, "gateway: there must be at most one of each IP family");
```

Comment:
This is good, but I wonder if it's maybe even possible to be more clear about the error here. Since I think at first we might be getting a lot of those!

Reply:
Thanks for the feedback! I tried to improve the message, let me know if you have something more in mind.